src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -655,7 +657,33 @@ private LocalSymbol GetDeclaredLocal(CSharpSyntaxNode declarationSyntax, SyntaxT
                 {
                     if (local.IdentifierToken == declaredIdentifier)
                     {
-                        return local;
+                        Debug.Assert(local is SourceLocalSymbol);
+                        LocalSymbol adjustedLocal;
+                        if (Compilation.NullableSemanticAnalysisEnabled)
+                        {
+                            if (!_analyzedVariableTypesOpt.TryGetValue(local, out adjustedLocal))
```

Comment:
>if (!_analyzedVariableTypesOpt.TryGetValue(local, out adjustedLocal)) [](start = 28, length = 69)

Please use: `_analyzedVariableTypesOpt.GetOrAdd(local, (local, arg) => { ... }, ...);`

Reply:
I can't. This project targets netstandard2.0, that api was added to netstandard until 2.1.

---
In reply to: [305589670](https://github.com/dotnet/roslyn/pull/37322#discussion_r305589670) [](ancestors = 305589670)

Reply:
An instance field could replace the lambda.
```
return _analyzedVariableTypesOpt.GetOrAdd(local, _getAnalyzedLocal);
```

---
In reply to: [306051195](https://github.com/dotnet/roslyn/pull/37322#discussion_r306051195) [](ancestors = 306051195,305589670)